### DISCUSSION TITLE
Pineapple doesn't recognize my wifi adapters

### DISCUSSION THREAD

**Discussion by Jonathan-R23** [xchwarze/wifi-pineapple-cloner#33] [Q&A]
Hi there, 

Hoping you are going well, in this case, I've built wifi pineapple nano for western N750, that is on the lis of devices available, by following the instructions:
1) I've installed the Openwrt factory image from the official web page https://downloads.openwrt.org/releases/19.07.7/targets/ath79/generic/openwrt-19.07.7-ath79-generic-wd_mynet-n750-squashfs-factory.bin
2) I built the WiFiPineapple
`tools/builder.sh mips nano imagebuilder/19.07.7-ath79-generic wd_mynet-n750`
The only warnings that I had in this process were:
`WARNING: Extractor.execute failed to run external extractor 'sasquatch -p 1 -le -d 'squashfs-root-0' '%e'': [Errno 2] No such file or directory: 'sasquatch', 'sasquatch -p 1 -le -d 'squashfs-root-0' '%e'' might not be installed correctly
WARNING: Extractor.execute failed to run external extractor 'sasquatch -p 1 -be -d 'squashfs-root-0' '%e'': [Errno 2] No such file or directory: 'sasquatch', 'sasquatch -p 1 -be -d 'squashfs-root-0' '%e'' might not be installed correctly
1598844       0x18657C        Squashfs filesystem, little endian, version 4.0, compression:xz, size: 13619288 bytes, 2541 inodes, blocksize: 262144 bytes, created: 2020-02-27 21:05:12`
But it successfully generated the files, one of them `openwrt-19.07.7-ath79-generic-wd_mynet-n750-squashfs-sysupgrade.bin` and I've updated the Openwrt with that file.
4) I successfully logged to the pinapple dashboard but I had problems with the wifi adapters, by looking the logs with `dmesg | grep ath9k` I found the following errors:
`[   11.736455] ath9k 18100000.wmac: Direct firmware load for ath9k-eeprom-ahb-18100000.wmac.bin failed wit                         h error -2
[   11.746921] ath9k 18100000.wmac: Falling back to user helper
[   16.589432] firmware ath9k-eeprom-ahb-18100000.wmac.bin: firmware_loading_store: map pages failed
[   16.599132] ath: phy0: Unable to load EEPROM file ath9k-eeprom-ahb-18100000.wmac.bin
[   16.607048] ath9k 18100000.wmac: failed to initialize device
[   16.612865] ath9k: probe of 18100000.wmac failed with error -22
[   16.625350] ath9k 0000:00:00.0: Direct firmware load for ath9k-eeprom-pci-0000:00:00.0.bin failed with                          error -2
[   16.635655] ath9k 0000:00:00.0: Falling back to user helper
[   16.868124] firmware ath9k-eeprom-pci-0000:00:00.0.bin: firmware_loading_store: map pages failed
[   16.877738] ath: phy1: Unable to load EEPROM file ath9k-eeprom-pci-0000:00:00.0.bin
[   16.885569] ath9k 0000:00:00.0: Failed to initialize device
[   16.891306] ath9k: probe of 0000:00:00.0 failed with error -22`

So, I'm stuck with this problem, hoping you can help me, I will appreciate a lot.

**Comment by wojtechit** [comment]
mine too can someone help please?

**Comment by m5kro** [comment]
https://github.com/Raxone/sasquatch_ubuntu_22.04

I had to use this to successfully build. For some reason it's not available through package repositories. You may need to spin up a vm or container with ubuntu 22.04.

**Comment by zeldoon** [comment]
did you try following from openwrt's help page  https://openwrt.org/packages/start

opkg list | grep -e <search>	Filter the list by a search term in the package name or its description.

opkg list | grep -e ath9k
ath9k-htc-firmware - 20190416-1 - AR9271/AR7010 firmware
kmod-ath9k - 4.14.221+4.19.161-1-1 - This module adds support for wireless adapters based on Atheros IEEE 802.11n AR5008 and AR9001 family of chipsets.
kmod-ath9k-common - 4.14.221+4.19.161-1-1 - Atheros 802.11n wireless devices (common code for ath9k and ath9k_htc)
kmod-ath9k-htc - 4.14.221+4.19.161-1-1 - This module adds support for wireless adapters based on Atheros USB AR9271 and AR7010 family of chipsets.
reghack - 1.0-4 - Reghack modifies the regulatory domain rules within the driver binaries with less restrictive ones. The current version also lifts the 5GHz radar channel restrictions in ath9k.
